JenkinsfileStressTests.groovy

```diff
@@ -0,0 +1,112 @@
+                maven 'maven'
+            }
+
+            stages {
+                stage('Compile') {
+                    steps {
+                        sh "mvn clean install -T1C -DskipTests"
+                    }
+                }
+                stage('Build image') {
+                    steps {
+                        script {
+                            sh "cp server/container/guice/cassandra-rabbitmq-guice/target/james-server-cassandra-rabbitmq-guice.jar dockerfiles/run/guice/cassandra-rabbitmq/destination"
+                            sh "cp -r server/container/guice/cassandra-rabbitmq-guice/target/james-server-cassandra-rabbitmq-guice.lib dockerfiles/run/guice/cassandra-rabbitmq/destination"
+                            sh "cp server/container/cli/target/james-server-cli.jar dockerfiles/run/guice/cassandra-rabbitmq/destination"
+                            sh "cp -r server/container/cli/target/james-server-cli.lib dockerfiles/run/guice/cassandra-rabbitmq/destination"
+                            sh 'cp server/protocols/jmap-integration-testing/rabbitmq-jmap-integration-testing/src/test/resources/keystore dockerfiles/run/guice/cassandra-rabbitmq/destination/conf'
+                            sh 'docker build -t james_run dockerfiles/run/guice/cassandra-rabbitmq'
+                        }
+                    }
+                }
+                stage('Start James') {
+                    steps {
+                        script {
+                            sh "cd /srv && sudo btrfs subvolume snapshot bench-snapshot bench-running-docker"
+                            sh 'docker run -d --name=cassandra -p 9042:9042 -p 9160:9160 -v /srv/bench-running-docker/cassandra:/var/lib/cassandra cassandra:3.11.3'
+                            sh 'docker run -d --name=elasticsearch -p 9200:9200 -p 9300:9300 -v /srv/bench-running-docker/elasticsearch:/usr/share/elasticsearch/data/elasticsearch elasticsearch:2.4.6'
+                            sh 'docker run -d --name=tika logicalspark/docker-tikaserver:1.19.1'
+                            sh 'docker run -d --name=swift -p 8080:8080 -v /srv/bench-running-docker/swift:/srv/1/node/sdb1 jeantil/openstack-keystone-swift:pike'
+                            sh 'docker run -d --name=rabbitmq -p 15672:15672 -p 5672:5672 rabbitmq:3.7.7-management'
+
+                            sh 'sleep 90'
+                            sh 'docker run -d --hostname HOSTNAME -p 25:25 -p 8000:8000 -p 110:110 -p 143:143 -p 465:465 -p 587:587 -p 993:993 --link cassandra:cassandra --link rabbitmq:rabbitmq --link elasticsearch:elasticsearch --link tika:tika --link swift:swift --name james_run -t james_run'
+                            sh 'sleep 90'
+                            try {
+                                def jamesCliWithOptions = 'java -jar /root/james-cli.jar -h 127.0.0.1 -p 9999'
+                                waitForCommandSuccess("docker exec james_run ${jamesCliWithOptions} listusers", maxRetries)
```

Comment:
you cannot use waitForCommandSuccess to restart james until it works?

Reply:
It fails too quickly

Reply:
maxretry is 120, so you are waiting ~120s (+ execution time). Instead of adding 2 90s sleeps, why don't you set maxRetry to 300?